client-runtime/src/main/java/com/microsoft/rest/v2/util/FlowableUtil.java

```diff
@@ -70,6 +72,70 @@ public static boolean isFlowableByteBuffer(TypeToken entityTypeToken) {
 
         });
     }
+    private static final class CountingOperator implements FlowableOperator<ByteBuffer, ByteBuffer> {
+        final long bytesExpected;
+        CountingOperator(long bytesExpected) {
+            if (bytesExpected < 0) {
+                throw new IllegalArgumentException("bytesExpected must be non-negative, but was " + bytesExpected);
+            }
+
+            this.bytesExpected = bytesExpected;
+        }
+
+        @Override
+        public Subscriber<? super ByteBuffer> apply(Subscriber<? super ByteBuffer> observer) {
+            final AtomicLong bytesRead = new AtomicLong();
+            final AtomicReference<Subscription> subscription = new AtomicReference<>();
```

Comment:
Put `subscription` as a non-volatile field in `Subscriber` because is part of the contract on upstream publisher that there is a happens-before relationship between `onSubscribe` and emissions.